vidarr-cardea/src/main/java/ca/on/oicr/gsi/vidarr/cardea/CardeaCasePriorityInput.java

```diff
@@ -0,0 +1,121 @@
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import io.undertow.Handlers;
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.StatusCodes;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.Optional;
+
+public class CardeaCasePriorityInput implements PriorityInput {
+
+  private static final HttpClient HTTP_CLIENT = HttpClient.newHttpClient();
+  private static final ObjectMapper MAPPER =
+      new ObjectMapper()
+          .registerModule(new Jdk8Module())
+          .configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true);
+  private int defaultPriority;
+  private BasicType schema;
+  private int ttl = 60;
+  private String baseUrl;
+  private KeyValueCache<String, Optional<Integer>> values;
+  private static final String casePriorityUrlFormat = "%s/cases/%s/priority";
+
+  @Override
+  public int compute(String workflowName, String workflowVersion, Instant created, JsonNode input) {
+    try {
+      return values.get(MAPPER.writeValueAsString(input)).orElse(defaultPriority);
+    } catch (JacksonException e) {
+      e.printStackTrace();
+      return defaultPriority;
+    }
+  }
+
+  public int getDefaultPriority() {
+    return defaultPriority;
+  }
+
+  public BasicType getSchema() {
+    return schema;
+  }
+
+  public int getTtl() {
+    return ttl;
+  }
+
+  public String getUrl() {
+    return baseUrl;
+  }
+
+  @Override
+  public Optional<HttpHandler> httpHandler() {
+    return Optional.of(Handlers.routing().delete("/", this::reset));
+  }
+
+  @Override
+  public BasicType inputFromSubmitter() {
+    return BasicType.STRING;
+  }
+
+  private void reset(HttpServerExchange exchange) {
+    exchange.setStatusCode(StatusCodes.NO_CONTENT);
+    values.invalidateAll();
+  }
+
+  public void setDefaultPriority(int defaultPriority) {
+    this.defaultPriority = defaultPriority;
+  }
+
+  public void setSchema(BasicType schema) {
+    this.schema = schema;
+  }
+
+  public void setTtl(int ttl) {
+    this.ttl = ttl;
+  }
+
+  public void setUrl(String url) {
+    this.baseUrl = url;
+  }
+
+  @Override
+  public void startup(String resourceName, String inputName) {
+    values =
+        new KeyValueCache<>(resourceName + " " + inputName, ttl, SimpleRecord::new) {
+          @Override
+          protected Optional<Integer> fetch(String body, Instant instant) throws Exception {
+            return HTTP_CLIENT
+                .send(
+                    HttpRequest.newBuilder(
+                            URI.create(
+                                String.format(
+                                    casePriorityUrlFormat,
+                                    baseUrl,
+                                    URLEncoder.encode(body, StandardCharsets.UTF_8)
```

Comment:
`body` is a bit ambiguous here, as it could be used to refer to a request body - but instead, it's referring to a case identifier.